extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1400,10 +1721,11 @@ void createNewTasks()
     }
   }
 
-  private void createKafkaTasksForGroup(int groupId, int replicas)
+  private void createKafkaTasksForGroup(int groupId, int replicas) throws JsonProcessingException
   {
     Map<Integer, Long> startPartitions = taskGroups.get(groupId).partitionOffsets;
-    Map<Integer, Long> endPartitions = new HashMap<>();
+    Map<Integer, Long> endPartitions = new HashMap<>(); // TODO if endOffsets were already set
```

Comment:
Can we resolve this TODO?

Reply:
removed the TODO, this case need not be handled because if endOffsets are set that means the TaskGroup has moved from active to publish pending state and supervisor never creates tasks for pending publish TaskGroup.